cdk/domino_cdk/aws_configurator.py

```diff
@@ -32,53 +10,17 @@ def __init__(self, scope: cdk.Construct, eks_cluster: eks.Cluster):
         self.install_calico()
 
     def install_calico(self):
-        # This produces an obnoxious diff on every subsequent run
-        # Using a helm chart does not, so we should switch to that
-        # However, we need to figure out how to get the helm chart
-        # accessible by the CDK lambda first. Not clear how to give
-        # s3 perms to it programmatically, and while ECR might be
-        # an option it also doesn't seem like there's a way to push
-        # the chart with existing api calls.
-        # Probably need to do some custom lambda thing.
-
-        crd_manifests = []
-        notcrd_manifests = []
-
-        for (name, url) in manifests:
-            filename = f"{name}.yaml"
-            if isfile(filename):
-                stream = Path(filename)
-            else:
-                # Something downstream will make this substitution anyway, cause fake diffs
-                stream = StringIO(sub(r'[“”]', '?', requests_get(url).text))
-
-            yaml = YAML(typ="safe")
-            loaded_manifests = list(yaml.load_all(stream))
-
-            crd_manifests += [crd for crd in loaded_manifests if crd["kind"] == "CustomResourceDefinition"]
-            notcrd_manifests += [notcrd for notcrd in loaded_manifests if notcrd["kind"] != "CustomResourceDefinition"]
-
-        # NB: be careful changing resource names or removing the manifests as prune is set by default
-        # and could inadvertantly remove other resources if deleted after the new manifest is created
-
-        # Split CRDs into multiple manifests so they don't go over the lambda limit of 262144 bytes
-        crds = []
-        for i, crd in enumerate(crd_manifests):
-            crds.append(
-                eks.KubernetesManifest(
-                    self.scope,
-                    # See above note about changing resource names
-                    "calico-crds" + ("" if i == 0 else str(i)),
-                    cluster=self.eks_cluster,
-                    manifest=[crd],
-                    overwrite=True,
-                )
-            )
-
-        if notcrd_manifests:
-            non_crds = eks.KubernetesManifest(
-                self.scope, "calico", cluster=self.eks_cluster, manifest=notcrd_manifests, overwrite=True
-            )
-
-            for crd in crds:
-                non_crds.node.add_dependency(crd)
+        eks.HelmChart(
```

Comment:
Does this work ok for upgrades? You don't need the same deletion as the TF side?

Reply:
I dont think it can as CF recreates the EKS stack.

Reply:
I was wrong, cdk attempts to upgrade but fails ti apply the helm release since it calico is installed. I thought CF would remove the resources it created.